src/software/new_geom/cubic_bezier_spline2d.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "software/new_geom/bezier_curve2d.h"
+#include "software/new_geom/spline2d.h"
+
+/**
+ * A Cubic Bezier spline is a C2 continuous spline that is formed by a series of bezier
+ * curves.
+ *
+ * Some useful resources:
+ * https://www.ibiblio.org/e-notes/Splines/Intro.htm
+ * https://www.ibiblio.org/e-notes/Splines/b-int.html
+ */
+class CubicBezierSpline2d : public Spline2d
+{
+   public:
+    CubicBezierSpline2d() = delete;
+
+    /**
+     * Create a CubicBezierSpline2d that:
+     *  - starts from `start` and is tangent to `start_vector` at the start
+     *  - ends at `end` and is tangent to `end_vector` at the end
+     *
+     *    ** EXAMPLE: **
+     *
+     *                           XXXXXXXXX
+     *        start_vector  XXXXX         XXXXX   end_vector
+     *              ^     XXX                 XXX     ^
+     *              |    XX                     XX    |
+     *              |   XX                       XX   |
+     *              |  XX                         XX  |
+     *              |  X                           X  |
+     *              | XX                           XX |
+     *              | X                             X |
+     *              | X                             X |
+     *              |XX                             XX|
+     *              |X                               X|
+     *              |X                               X|
+     *              |X                               X|
+     *       start  +                                 +  end
+     *
+     *
+     * @param start_point The first point on the spline
+     * @param start_vector The tangent vector at the start of the spline
```

Comment:
Does the vector magnitude mean anything?

Reply:
hmm, yes it does. 

If we have a curve with control points P0-P4 inclusive, then (from wikipedia):
> The distance between P1 and P2 determines "how far" and "how fast" the curve moves towards P1 before turning towards P2.

It's difficult to articulate this, but it is important that we convey it. Any suggestions?

Reply:
Maybe something along the lines of 'vector magnitude 'stretches' the curve farther in that direction'

Reply:
I'd be fine with referencing the control points, but since this is a "Spline" class maybe we shouldn't?

Alternative suggestion:

"The direction of the start and end vectors determine the direction of the curve at the start and end points. The magnitude of these vectors determine how far the curve is stretched in the direction of the vector before connecting with the intermediate points"

Reply:
bumping this

Reply:
nice little blurb, added